src/main/webapp/resources/js/pages/admin/components/remote-connections/RemoteApiStatus.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState } from "react";
+import { Alert, Button } from "antd";
+import { checkConnectionStatus } from "../../../../apis/remote-api/remote-api";
+import { setBaseUrl } from "../../../../utilities/url-utilities";
+import { IconLoading, IconLogin } from "../../../../components/icons/Icons";
+import { SPACE_XS } from "../../../../styles/spacing";
+import { formatInternationalizedDateTime } from "../../../../utilities/date-utilities";
+
+/**
+ * React component to render the status of a Remote API.
+ * If the API is not connected it will present the user a button allowing
+ * them to connect.
+ * @param {object} api - details about the remote API
+ * @param {function} onConnect - what do when the connection is made.
+ * @returns {*}
+ * @constructor
+ */
+export function RemoteApiStatus({ api, onConnect = () => {} }) {
+  const [loading, setLoading] = useState(true);
+  const [expiration, setExpiration] = useState(undefined);
+
+  useEffect(checkApiStatus, []);
+
+  useEffect(() => {
+    // Listen for a remote api connection
+    window.addEventListener("message", updateRemoteApi, false);
+    return () => window.removeEventListener("message", updateRemoteApi);
+  }, []);
+
+  useEffect(() => checkApiStatus(), [api.id]);
+
+  function checkApiStatus() {
+    setLoading(true);
+    checkConnectionStatus({ id: api.id }).then((data) => {
+      setLoading(false);
+      setExpiration(data);
+      data && onConnect();
+    });
+  }
+
+  function updateRemoteApi(event) {
+    if (
+      event.origin === window.location.origin &&
+      event.data === "remote_api_connect"
+    ) {
+      checkApiStatus();
+    }
+  }
+
+  /**
+   * This will open a popup window with the Oauth for the Remote API
+   */
+  function updateConnectionStatus() {
+    const w = 600;
+    const h = 400;
+
+    const left = screen.width / 2 - w / 2;
+    const top = screen.height / 2 - h / 2;
+    window.open(
+      setBaseUrl(`remote_api/connect/${api.id}`),
+      "",
+      `toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width=${w}, height=${h}, top=${top}, left=${left}`
+    );
+  }
+
+  return loading ? (
+    <span>
+      <IconLoading style={{ marginRight: SPACE_XS }} />
+      {i18n("RemoteApi.checking")}
+    </span>
+  ) : (
+    <div>
+      {expiration ? (
+        <Alert
+          className="t-remote-status-connected"
+          message={i18n(
+            "RemoteApi.connected",
```

Comment:
Issue with this new expiration message.  It doesn't actually check if the connection is valid before displaying this message.  There could be a case where the token is revoked but it just displays the expires message instead of validating the token.

Steps to reproduce:
1. Create client, remote api, and connect
2. Go to clients and revoke the token
3. Look at remote api pages and it just says "expires whenever" and doesn't actually check if its valid

In this case it should display the "connect" button again so users can get their new token.